### ISSUE TITLE
untranslatable string: global variable

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#3962]
![darkforecast](https://user-images.githubusercontent.com/21158813/53685201-e78f9680-3d17-11e9-8106-97898d43a17d.png)

I grepped recently for it, unless I didn't miss anything it's not translatable.

**Comment by jostephd** [comment]
It's this:

https://github.com/wesnoth/wesnoth/blob/49612b89e1aa4c89365a0e3e6080ced8476098ab/src/persist_var.cpp#L52-L55

https://github.com/wesnoth/wesnoth/blob/49612b89e1aa4c89365a0e3e6080ced8476098ab/src/synced_user_choice.cpp#L326-L331

There are some other relevant strings, for example:

https://github.com/wesnoth/wesnoth/blob/49612b89e1aa4c89365a0e3e6080ced8476098ab/src/actions/advancement.cpp#L220-L223

Do you want just to add `_()` around the first one? Or do we need a larger fix along the lines of https://forums.wesnoth.org/viewtopic.php?f=8&t=11445?

**Comment by gfgtdf** [comment]
'a global variable' is probably a little too technical for a user visiable message in the first place. Maybe we should allow the wml code to specify that message (adding a description= key in [get_global_variable]) then the wml author (that knowns which information exqactly is stored in that variable) can speficy the message.
(This one in the first post seems to come from a non-mainline wml code so i cannot give a concreate exampel here)

**Comment by sevu** [comment]
'a global variable' is indeed to technical. I didn't find a better term though either.
I think might ` _( )` be good enough if we find a better wording. Or maybe rewording the whole statement?

This case is actually a mainline scenario, 2P Dark Forecast, just played with other factions (enemies are still mainline). I saw this message a few weeks ago for a moment, this time it showed long enough to make a screenshot the slow way, finding the snipping tool in the app menu, changing the options, selecting an area...

Side 1 is actually an AI side here, running on the remote host.

**Comment by sevu** [comment]
Found one more, as spectator in a randomly generated map at gamestart.
![input](https://user-images.githubusercontent.com/21158813/53687948-ecfed800-3d3b-11e9-96a9-c3451f9541c6.png)

**Comment by gfgtdf** [comment]
> Found one more, 


I think that's this one https://github.com/wesnoth/wesnoth/blob/master/src/scripting/game_lua_kernel.cpp#L2887 (which is maked as translatable)


This one can easily tranlated as "eine Eingabe" in German

**Comment by gfgtdf** [comment]
Make this is an erro in the gettet scanner? Coudl you please check whether the string in that .lua file is present in the pot files?

**Comment by CelticMinstrel** [comment]
1. Even if you add a key to specify the description, you'll want to have a default to fall back on if the user-specified description is missing. I think something about "loading data" would be fine, as global variables are all about client-specific data stored on their computers.
2. The key should probably be called `wait_description` for consistency with `[message]` and possibly other places.

**Comment by jostephd** [comment]
@gfgtdf My money is on this one: https://github.com/wesnoth/wesnoth/blob/5cdd51d0e2c791b9d08e9dd02cfa04caf128dc5e/src/synced_user_choice.hpp#L38

I assume it should become `_("waiting for^input")` or something like that, so translators would know to translate it into the appropriate grammatical case? And likewise the other overrides of that virtual method.

**Comment by CelticMinstrel** [comment]
That's a good idea.

**Comment by gfgtdf** [comment]
> @gfgtdf My money is on this one:

Wait isn't that just an unused fallback? I thought all subclasses overwrite that one.

**Comment by sevu** [comment]
Good idea with the prefix. Could we also add a translation hint? Sth. like
`# po: In networked games, this text is shown on the map while waiting for $desc from another player. Don't end the text with a punctuation sign.`

And also for the other ones:
`# po: In networked games, when one player has the choice between multiple advancements of a unit, this text is sent to other players. It will be displayed within a message.`

`# po: In networked games, this text is shown for other clients, while they wait to receive the content of a global variable from another player. This text will be embedded into a sentence.`

`# po: In networked games, this text is shown for other clients, when ...... What kind of input we wait for, [message]s with options? `
( Syntax for translation comment was another one for C++ )

**Comment by sevu** [comment]
Reopening for 
 https://github.com/wesnoth/wesnoth/blob/5cdd51d0e2c791b9d08e9dd02cfa04caf128dc5e/src/synced_user_choice.hpp#L38
> 12:46] jyrkive: IIRC, gettext.hpp must not be used in a header.
> [12:46] jyrkive: Thus, that string cannot be marked translatable.

**Comment by CelticMinstrel** [comment]
Does that commit close this again?

**Comment by sevu** [comment]
Yes

**Comment by gfgtdf** [comment]
Why was this closed, Is jyrkives comment no longer valid?

**Comment by sevu** [comment]
It compiles.

More context:
> [13:38] BOTIRC Bridge: <celmin|sleep> You can use gettext.hpp in headers, but it's better to avoid it.
[13:38] BOTIRC Bridge: <celmin|sleep> Originally gettext didn't even parse headers, though I think that was changed at some point.

> [14:20] loonycyborg: I can see one problem with using translateable strings in headers
[14:20] loonycyborg: to which textdomain will they apply?
[14:23] jyrkive: Ah, right. Setting the textdomain in a header is a bad idea, and thus it will depend on where the header is #included.
[14:23] jyrkive: The same string may even easily end up in multiple domains.

They are all in the `wesnoth` textdomain

**Comment by CelticMinstrel** [comment]
Unless the string is in a template class, I would recommend refactoring the code so that translatable strings are not present in headers. In this specific case it looks like that should be trivial (unless it _is_ a template class, I didn't look that closely).
